tests/components/iotawatt/test_config_flow.py

```diff
@@ -0,0 +1,93 @@
+"""Test the IoTawatt config flow."""
+from unittest.mock import patch
+
+from homeassistant import config_entries, setup
+from homeassistant.components.iotawatt.config_flow import CannotConnect, InvalidAuth
+from homeassistant.components.iotawatt.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import RESULT_TYPE_CREATE_ENTRY, RESULT_TYPE_FORM
+
+
+async def test_form(hass: HomeAssistant) -> None:
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == config_entries.SOURCE_USER
+
+    with patch(
+        "homeassistant.components.iotawatt.async_setup_entry",
+        return_value=True,
+    ), patch(
+        "homeassistant.components.iotawatt.config_flow.validate_input",
+        return_value={"title": "iotawatt"},
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "name": "iotawatt",
+                "host": "1.1.1.1",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_invalid_auth(hass: HomeAssistant) -> None:
+    """Test we handle invalid auth."""
+
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == config_entries.SOURCE_USER
+
+    with patch(
+        "homeassistant.components.iotawatt.async_setup_entry",
+        return_value=True,
+    ), patch(
+        "homeassistant.components.iotawatt.config_flow.validate_input",
+        side_effect=InvalidAuth,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "name": "iotawatt",
+                "host": "1.1.1.1",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == RESULT_TYPE_FORM
+    assert len(mock_setup_entry.mock_calls) == 1
+
+    result = await hass.config_entries.flow.async_init(
```

Comment:
You are not asserting anything on these results?

Reply:
Removed ```result```.

Reply:
Though this test needs to check the error type. The error type for the ```test_form_invalid_auth``` should contain an invalid auth error but is not. Will have to look into this, probably the root cause is an exciting Issue [#21](https://github.com/gtdiehl/iotawatt_ha/issues/21)

Reply:
The authentication problem during setup is now resolved with the latest code changes.